system/usbhcd.c

```diff
@@ -338,6 +338,53 @@ static bool configure_cp210x(const usb_hcd_t *hcd, const usb_ep_t *ep0, int inte
     return true;
 }
 
+#define pl2303_vendor_read(value) do { \
+    build_setup_packet(&setup_pkt, 0xc0, 0x01, value, 0, 1); \
+    if (!hcd->methods->get_data_request(hcd, ep0, &setup_pkt, &buf1, 1)) { \
+        return false; \
+    } \
+  } while (0)
+
+#define pl2303_vendor_write(value, index) do { \
+    build_setup_packet(&setup_pkt, 0x40, 0x01, value, index, 0); \
+    if (!hcd->methods->setup_request(hcd, ep0, &setup_pkt)) { \
+        return false; \
+    } \
+  } while (0)
+
+static bool configure_pl2303(const usb_hcd_t *hcd, const usb_ep_t *ep0)
+{
+    usb_setup_pkt_t setup_pkt;
+    uint8_t buf1;
+
+    // Magic sequence
+    pl2303_vendor_read(0x8484);
```

Comment:
At least the setup packets containing 0x8484 and 0x8383 should be extracted under `static const` form, so as to avoid repeated on-the-fly generation of the packet data. But AFAICT, all of these setup packets containing constant data could be made `static const` as well.

Reply:
Thanks for the hint. For now I am deliberately using the same function names and constants as in the kernel source while we pick the parts that we want to use and how much we want to support, and any optimization can follow. I have also refrained from any reorganization and renaming of the memtest86plus code base. If you decide to go for this we'll may want to organize the USB code, like a new folder for USB or for USB-serial and separate out real keyboard handling etc. I don't know if you would prefer to merge then reorganize or the other way around.